src-extras/Database/LSMTree/Extras/UTxO.hs

```diff
@@ -0,0 +1,77 @@
+{-# LANGUAGE DeriveAnyClass             #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE DerivingStrategies         #-}
+{-# LANGUAGE GeneralisedNewtypeDeriving #-}
+{-# LANGUAGE NamedFieldPuns             #-}
+
+module Database.LSMTree.Extras.UTxO (
+    UTxOKey (..)
+  , UTxOValue (..)
+  , UTxOBlob (..)
+  ) where
+
+import           Control.DeepSeq
+import qualified Data.ByteString as BS
+import           Data.WideWord.Word256
+import           Database.LSMTree.Extras.Orphans ()
+import           Database.LSMTree.Internal.Serialise
+import           Database.LSMTree.Internal.Serialise.Class as Class
+import           GHC.Generics
+import           System.Random
+import           Test.QuickCheck
+
+{-------------------------------------------------------------------------------
+  UTxO keys
+-------------------------------------------------------------------------------}
+
+-- | A model of a UTxO key (256-bit hash)
+newtype UTxOKey = UTxOKey Word256
+  deriving stock (Show, Generic)
+  deriving newtype ( Eq, Ord, NFData
+                   , SerialiseKey
+                   , Num, Enum, Real, Integral
+                   )
+  deriving anyclass Uniform
+
+instance Arbitrary UTxOKey where
+  arbitrary = UTxOKey <$>
+      (Word256 <$> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary)
+  shrink (UTxOKey w256) = [
+        UTxOKey w256'
+      | let i256 = toInteger w256
+      , i256' <- shrink i256
+      , toInteger (minBound :: Word256) <= i256'
+      , toInteger (maxBound :: Word256) >= i256'
+      , let w256' = fromIntegral i256'
+      ]
+
+{-------------------------------------------------------------------------------
+  UTxO values
+-------------------------------------------------------------------------------}
+
+-- | A model of a UTxO value (512-bit)
+data UTxOValue = UTxOValue {
```

Comment:
Also the the value is 60 bytes; this overshoots.

Reply:
Indeed.